# Exploring Sea Ice Concentration Datasets

In [3]:
import xarray as xr
def explore_nc(f):
    ds = xr.open_dataset(f)
    return ds

**Met Office Hadley Centre Sea Ice and Sea Surface Temperature dataset**

In [4]:
file_path_MetO = "/Users/bean/Documents/Doctorate/Fall_24/Mokbel8735_Adv_Databases/climate_data_integration/data/raw/HadISST.2.2.0.0_sea_ice_concentration.nc"
ds = explore_nc(file_path_MetO)
ds

<xarray.Dataset> Size: 531MB
Dimensions:    (time: 2048, nv: 2, latitude: 180, longitude: 360)
Coordinates:
  * time       (time) datetime64[ns] 16kB 1850-01-16T11:59:59.505615234 ... 2...
  * latitude   (latitude) float32 720B 89.5 88.5 87.5 86.5 ... -87.5 -88.5 -89.5
  * longitude  (longitude) float32 1kB 0.5 1.5 2.5 3.5 ... 357.5 358.5 359.5
Dimensions without coordinates: nv
Data variables:
    time_bnds  (time, nv) float32 16kB ...
    sic        (time, latitude, longitude) float32 531MB ...
Attributes:
    Title:                      Monthly 1 degree version of HadISST.2.2.0.0
    source:                     Met Office
    reference:                  update of Titchner and Rayner 2015 DOI: 10.10...
    Conventions:                CF-1.0
    history:                    5/10/2020 converted to netcdf
    supplementary_information:  contact holly.titchner@metoffice.gov.uk or jo...

The latitude dimension goes from 89.5 to -89.5 degrees, increasing by 1 degree each time. 

The longitude dimensio goes from 0.5 to 359.5 degrees, increasing by 1 degree each time.

The time dimension goes from January 16, 1850 to August 16, 2020, increasing by 1 month each time. Note the measurements are taken at various hours of the day.

**COBE-SST 2 and Sea Ice data**

In [5]:
file_path_NOAA = "/Users/bean/Documents/Doctorate/Fall_24/Mokbel8735_Adv_Databases/climate_data_integration/data/raw/icec.mon.mean.nc"
ds = explore_nc(file_path_NOAA)
ds

<xarray.Dataset> Size: 529MB
Dimensions:  (lat: 180, lon: 360, time: 2040)
Coordinates:
  * lat      (lat) float32 720B 89.5 88.5 87.5 86.5 ... -86.5 -87.5 -88.5 -89.5
  * lon      (lon) float32 1kB 0.5 1.5 2.5 3.5 4.5 ... 356.5 357.5 358.5 359.5
  * time     (time) datetime64[ns] 16kB 1850-01-01 1850-02-01 ... 2019-12-01
Data variables:
    icec     (time, lat, lon) float32 529MB ...
Attributes:
    title:            created 05/2014 from data provided by JRA
    history:          Created 12/2012 from data obtained from JRA by ESRL/PSD
    platform:         Analyses
    citation:         Hirahara, S., Ishii, M., and Y. Fukuda,2014: Centennial...
    institution:      NOAA ESRL/PSD
    Conventions:      CF-1.2
    dataset_title:    COBE-SST2 Sea Surface Temperature and Ice
    original_source:  https://climate.mri-jma.go.jp/pub/ocean/cobe-sst2/
    References:       https://www.psl.noaa.gov/data/gridded/cobe2.html

The latitude dimension goes from 89.5 to -89.5 degrees, increasing by 1 degree each time. 

The longitude dimensio goes from 0.5 to 359.5 degrees, increasing by 1 degree each time.

The time dimension goes from January 01, 1850 to December 01, 2019, increasing by 1 month each time. All the measurements are taken at the same hour (00:00).

## TODO

* Check that the pipeline steps are correct
* Try to match the time dimensions even though they are taken at different days
* Finish matching function
* Finish pipeline -- get user input
* Check if variable values are the same (they shouldn't be since they're taken at different days of the month)
* Try to aggregate the datasets various ways
    1. No scaling
    2. Scaling different ways (to model confidence in data)
* Write report:
    * Why is this a difficult problem

Aggregating:

1. No scaling
2. MetO = 0.2, NOAA = 0.8
3. NOAA = 0.2, MetO = 0.8
4. mean, max, min (no scaling)
5. First 2/3 of NOAA, last 2/3 of MetO, mean <-- show how you can get 100